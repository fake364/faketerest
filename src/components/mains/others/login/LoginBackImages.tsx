import React, { useEffect } from 'react';
import { SLIDER_THEMES } from '../../../../common/enums/slider_themes';
import CommonUtils from '../../../../common/utils/creation-utils/arrays/common';
import Image from 'next/image';
import clsx from 'clsx';

type Props = {};

const LoginBackImages: React.FC<Props> = ({ children }) => {
  useEffect(() => {
    document.body.style.overflow = 'hidden';
  }, []);

  return (
    <div className="flex flex-wrap gap-[16px] px-[52px] py-[18px]">
      <div className="absolute w-full h-full bg-[rgba(0,0,0,0.6)] top-0 left-0 z-10" />
      {children}
      {CommonUtils.arrayOfImagesByType(SLIDER_THEMES.HOME_DECOR).map(
        (image) => (
          <div className="flex-1 relative basis-[200px] h-[400px]">
            <Image
              className={clsx('rounded-[16px]')}
              src={image}
              layout={'fill'}
              objectFit="cover"
            />
          </div>
        )
      )}
    </div>
  );
};

export default LoginBackImages;
