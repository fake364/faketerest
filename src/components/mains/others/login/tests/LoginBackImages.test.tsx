import { mount, shallow } from 'enzyme';
import { expect, jest } from '@jest/globals';
import React from 'react';
import LoginBackImages from '../LoginBackImages';
import CommonUtils from '../../../../../common/utils/creation-utils/arrays/common';

jest.mock('next/image', () => (props) => <div {...props} />);

describe('LoginBackImages', () => {
  beforeEach(() => {
    jest
      .spyOn(CommonUtils, 'arrayOfImagesByType')
      .mockReturnValue(['1', '2', '3']);
  });

  it('should render container properly', async () => {
    const component = await shallow(<LoginBackImages>Test</LoginBackImages>);
    expect(component).toMatchSnapshot();
  });

  it('should set body overflow to hidden', async () => {
    await mount(<LoginBackImages>Test</LoginBackImages>);
    expect(document.body.style.overflow).toBe('hidden');
  });
});
