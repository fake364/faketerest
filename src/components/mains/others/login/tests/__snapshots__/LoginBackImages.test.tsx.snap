// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginBackImages should render container properly 1`] = `
<div
  className="flex flex-wrap gap-[16px] px-[52px] py-[18px]"
>
  <div
    className="absolute w-full h-full bg-[rgba(0,0,0,0.6)] top-0 left-0 z-10"
  />
  Test
  <div
    className="flex-1 relative basis-[200px] h-[400px]"
  >
    <Component
      className="rounded-[16px]"
      layout="fill"
      objectFit="cover"
      src="1"
    />
  </div>
  <div
    className="flex-1 relative basis-[200px] h-[400px]"
  >
    <Component
      className="rounded-[16px]"
      layout="fill"
      objectFit="cover"
      src="2"
    />
  </div>
  <div
    className="flex-1 relative basis-[200px] h-[400px]"
  >
    <Component
      className="rounded-[16px]"
      layout="fill"
      objectFit="cover"
      src="3"
    />
  </div>
</div>
`;
